src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -538,43 +538,15 @@ void QgsTemporalControllerWidget::setTimeStep( const QgsInterval &timeStep )
   if ( ! timeStep.isValid() || timeStep.seconds() <= 0 )
     return;
 
-  // Search the time unit the most appropriate :
```

Comment:
When this method was implemented, the aim was to provide an adapted time step (duration and unit) when loading mesh layer (https://github.com/qgis/QGIS/pull/36949/files) and the method was used only for that and was called only when first mesh layer is loaded. 
Since this method is now called when frame duration is changed (even by user). I think the problem is not what  this original method do internally but it could be maybe better to create another method that will be called when frame duration changed. So, maybe, let this original method like before, and create another to change the time step when frame duration change.
Indeed, I think I will be quite sad if this  there is no this "unit guessing" anymore (and maybe also other users of mesh layers).

Reply:
> So, maybe, let this original method like before, and create another to change the time step when frame duration change

Sure @vcloarec , I will create another one then.